sympy/stats/crv_types.py

```diff
@@ -2195,6 +2196,113 @@ def Normal(name, mean, std):
 
+
+#-------------------------------------------------------------------------------
+# Inverse Normal distribution ----------------------------------------------------------
+
+
+class NormalInverseDistribution(SingleContinuousDistribution):
+    _argnames = ('mean', 'shape')
+
+    @property
+    def set(self):
+        return Interval(0, oo)
+
+    @staticmethod
+    def check(mean, shape):
+        _value_check(shape > 0, "Shape parameter must be positive")
+        _value_check(mean > 0, "Mean must be positive")
+
+    def pdf(self, x):
+        return exp(-self.shape*(x - self.mean)**2 / (2*x*self.mean**2)) * sqrt(self.shape/((2*pi*x**3)))
+
+    def sample(self):
+        scipy = import_module('scipy')
+        if scipy:
+            from scipy.stats import invgauss
+            return invgauss.rvs(float(self.mean/self.shape), 0, float(self.shape))
+        else:
+            raise NotImplementedError('Sampling the Inverse Normal Distribution requires Scipy.')
+
+    def _standardNormal_cdf(self, x):
+        return erf(sqrt(2)*(x - S.One)/2)/2 + S.Half
+
+    def _cdf(self, x):
+        mean, shape = self.mean, self.shape
+
+        first_term = self._standardNormal_cdf(sqrt(shape/x) * ((x/mean - S.One)))
+        second_term = exp(2*shape/mean) * self._standardNormal_cdf(-sqrt(shape/x)*(x/mean + S.One))
+
+        return  first_term + second_term
+
+    def _characteristic_function(self, t):
+        mean, shape = self.mean, self.shape
+        return exp((shape/mean)*(1 - sqrt(1 - (2*mean**2*I*t)/shape)))
+
+    def _moment_generating_function(self, t):
+        mean, shape = self.mean, self.shape
+        return exp((shape/mean)*(1 - sqrt(1 - (2*mean**2*t)/shape)))
+
+
+def NormalInverse(name, mean, shape):
+    r"""
+    Create a continuous random variable with an Inverse Normal distribution.
+
+    The density of the Inverse Normal distribution is given by
+
+    .. math::
+        f(x) := \sqrt(\frac{\lambda}{2\pi x^3}) e^{-\frac{\lambda(x-\mu)^2}{2x\mu^2}}
+
+    Parameters
+    ==========
+
+    mu : Positive number representing the mean
+    lambda : Positive number representing the shape parameter
+
+    Returns
+    =======
+
+    A RandomSymbol.
+
+    Examples
+    ========
+
+    >>> from sympy.stats import NormalInverse, density, E, std, cdf, skewness
+    >>> from sympy import Symbol, simplify, factor, together, factor_terms
+
+    >>> mu = Symbol("mu", positive=True)
+    >>> l = Symbol("lambda", positive=True)
+    >>> z = Symbol("z", positive=True)
+    >>> y = Symbol("y")
+    >>> p = Symbol("p")
+    >>> X = NormalInverse("x", mu, l)
+
+    >>> density(X)(z)
+    sqrt(2)*sqrt(lambda)*exp(-lambda*(-mu + z)**2/(2*mu**2*z))/(2*sqrt(pi)*z**(3/2))
+
+    >>> simplify(cdf(X))(z)
+    (1 - erf(sqrt(2)*(sqrt(lambda)*(mu + z) + mu*sqrt(z))/(2*mu*sqrt(z))))*exp(2*lambda/mu)/2 -
+         erf(sqrt(2)*(sqrt(lambda)*(mu - z) + mu*sqrt(z))/(2*mu*sqrt(z)))/2 + 1/2
+
+    >>> simplify(skewness(X))
```

Comment:
Is `simplify` required here?
Also, I'd suggest minimising it's use in library code as it can be slow.

Reply:
By using `simplify` I am able to get the result as given in [wiki](https://en.wikipedia.org/wiki/Inverse_Gaussian_distribution). I will remove it and rewrite the answer.

Reply:
Without using `simplify` I am getting the following result: `(2*mu**3 - 3*mu*(lambda*mu**2 + mu**3)/lambda + (lambda**2*mu**3 + 3*lambda*mu**4 + 3*mu**5)/lambda**2)/(-mu**2 + (lambda*mu**2 + mu**3)/lambda)**(3/2)`. Should I keep this or the previous one?

Reply:
`ratsimp` could be used. Also, you can define the symbol as `std = Symbol('lamda')`. Note that, in SymPy `Symbol('lamda')` also prints as greek symbol λ. (This is done to avoid confusion with `lambda`, used for creation of anonymous functions in Python).

Reply:
`simplify` is an expensive method that should be avoided in tests to save time. In most cases a more specific simplification like `expand` will suffice.